pkg/chunk/table_manager.go

```diff
@@ -540,3 +543,10 @@ func ExpectTables(ctx context.Context, client TableClient, expected []TableDesc)
 
 	return nil
 }
+
+func (m *TableManager) validateClient() bool {
+	if reflect.TypeOf(m.client).String() == "*local.tableClient" {
+		return false
+	}
+	return true
+}
```

Comment:
If this implementation is OK, will add a small test for this. Let me know if we can do it in any other better way.